### DISCUSSION TITLE
Will Enhancement CSV  files be cached? When querying, is the data retrieved from the disk or from memory? If it's from memory, how can hot reload be achieved when the CSV file is updated?

### DISCUSSION THREAD

**Discussion by Rentu** [vectordotdev/vector#22086] [Q&A] [answered]
Will Enhancement CSV  files be cached? When querying, is the data retrieved from the disk or from memory? If it's from memory, how can hot reload be achieved when the CSV file is updated?

**Comment by jszwedko** [comment, marked as answer]
Hi @Rentu !

The CSV is loaded at start-up into memory. It is only reloaded if a SIGHUP is issued to Vector to trigger a reload. There is an issue to enhance Vector to watch the file for changes and trigger a reload: https://github.com/vectordotdev/vector/issues/11817

See also: https://github.com/vectordotdev/vector/discussions/19782

**Reply by Rentu** [reply]
Ok, tks
